lib/AuroraLip/Common/FormatInfo.cs

```diff
@@ -202,7 +202,7 @@ public string GetTypName()
         /// <inheritdoc />
         public virtual bool Equals(FormatInfo other)
         {
-            if (Identifier != null)
+            if (Identifier != null && other.Identifier != null)
             {
                 return Identifier.Equals(other.Identifier);
```

Comment:
The problem will be that `other` is null, we should check this for null.

`                return Identifier.Equals(other?.Identifier);`
`                return Extension.ToLower() == other?.Extension.ToLower();`

Reply:
ok i see the mistake i forgot the null check at class `Identifier` && `Identifier64`
        `public bool Equals(IIdentifier other) => other.AsSpan().SequenceEqual(AsSpan());`
to 
        `public bool Equals(IIdentifier other) => other != null && other.AsSpan().SequenceEqual(AsSpan());`

my fault

Reply:
That works too.  Would you prefer that?

Reply:
yes because it fixes the actual error

Reply:
This does too though.. but I can fix it there instead.

Reply:
Ah, I see what you're saying.  It's more "direct".  Yep!